### ISSUE TITLE
Grammar not correctly recognizing doc comment types in some cases

### ISSUE THREAD

**Issue by abgivant** [Microsoft/TypeScript-TmLanguage#300]
When using the new grammar in VS Code, highlighting of doc comments fails if either the `Class.<type>` syntax is used (this is valid in [JSDoc](http://usejsdoc.org/tags-type.html)), or if types are nested more than one level deep. It also currently does not recognize explicitly documenting the properties of an object.

![image](https://cloud.githubusercontent.com/assets/3431954/19252666/9f2ecf80-8f0b-11e6-8ae7-46d1db3e0740.png)

Are there currently plans to support any of these notations?

Thanks!

**Comment by sheetalkamat** [comment]
@abgivant I have added support for the ```{Array.<string>}``` but support for ```{Promise<Array<string>>}``` and ```{{prop: string}}``` is still not added. We use the jsdoc grammar from [atom/language-javascript repo](https://github.com/atom/language-javascript) and would be waiting on their implementation of it.

**Comment by abgivant** [comment]
Thanks! 

And that's fine. I can open a new issue in that repo with the remaining two issues so you can close this if you'd like.

**Comment by abgivant** [comment]
Opened https://github.com/atom/language-javascript/issues/451 in the Atom language repo.
